config/opal_check_pmi.m4

```diff
@@ -180,14 +183,17 @@ AC_DEFUN([OPAL_CHECK_PMIX_LIB],[
           [$3
            # add the new flags to our wrapper compilers
            AS_IF([test "$pmix_ext_install_incdir" != "/usr" && test "$pmix_ext_install_incdir" != "/usr/include"],
-                 [pmix_pmix4x_WRAPPER_EXTRA_CPPFLAGS="-I$pmix_ext_install_incdir"])
+                 [pmix_external_WRAPPER_EXTRA_CPPFLAGS="-I$pmix_ext_install_incdir"])
            AS_IF([test "$pmix_ext_install_libdir" != "/usr" && test "$pmix_ext_install_libdir" != "/usr/include"],
                  [pmix_external_WRAPPER_EXTRA_LDFLAGS="-L$pmix_ext_install_libdir"])
-           pmix_external_WRAPPER_EXTRA_LIBS=-lpmix],
-          [CPPFLAGS=$opal_external_pmix_save_CPPFLAGS
-           LDFLAGS=$opal_external_pmix_save_LDFLAGS
-           LIBS=$opal_external_pmix_save_LIBS
-           $4])
+                  pmix_external_WRAPPER_EXTRA_LIBS=-lpmix],
+          [$4])
+
+    AC_SUBST(pmix_external_WRAPPER_EXTRA_LDFLAGS)
+    AC_SUBST(pmix_external_WRAPPER_EXTRA_LIBS)
+dnl swap back in original LDFLAGS, LIBS to avoid messing up subsequent configury checks
```

Comment:
Should we also restore CFLAGS?

Reply:
I tried that but you end up having to change several makefiles.  Didn't seem worth it.

Reply:
Yes, it is worth it.  We can't pollute the top-level CPPFLAGS.

You need to set PMIx-related CPPLAGS only where it's relevant to use them.

Sorry.  😦